src/jvm/test/resources/expected_logs/bounded_queue_incorrect_results.txt

```diff
@@ -0,0 +1,424 @@
+|                                                                                             |     dequeue(): null at QueueAdaptor.dequeue(BoundedQueueTest.kt:99)             |            |
+|                                                                                             |     concurrentEnqueues.READ: 1 at QueueAdaptor.dequeue(BoundedQueueTest.kt:105) |            |
+|                                                                                             |     dequeue(): null at QueueAdaptor.dequeue(BoundedQueueTest.kt:99)             |            |
+|                                                                                             |     concurrentEnqueues.READ: 1 at QueueAdaptor.dequeue(BoundedQueueTest.kt:105) |            |
+|                                                                                             |     dequeue(): null at QueueAdaptor.dequeue(BoundedQueueTest.kt:99)             |            |
+|                                                                                             |     concurrentEnqueues.READ: 1 at QueueAdaptor.dequeue(BoundedQueueTest.kt:105) |            |
+|                                                                                             |     dequeue(): null at QueueAdaptor.dequeue(BoundedQueueTest.kt:99)             |            |
+|                                                                                             |     concurrentEnqueues.READ: 1 at QueueAdaptor.dequeue(BoundedQueueTest.kt:105) |            |
+|                                                                                             |     dequeue(): null at QueueAdaptor.dequeue(BoundedQueueTest.kt:99)             |            |
+|                                                                                             |     concurrentEnqueues.READ: 1 at QueueAdaptor.dequeue(BoundedQueueTest.kt:105) |            |
+|                                                                                             |     dequeue(): null at QueueAdaptor.dequeue(BoundedQueueTest.kt:99)             |            |
+|                                                                                             |     concurrentEnqueues.READ: 1 at QueueAdaptor.dequeue(BoundedQueueTest.kt:105) |            |
+|                                                                                             |     dequeue(): null at QueueAdaptor.dequeue(BoundedQueueTest.kt:99)             |            |
+|                                                                                             |     concurrentEnqueues.READ: 1 at QueueAdaptor.dequeue(BoundedQueueTest.kt:105) |            |
+|                                                                                             |     dequeue(): null at QueueAdaptor.dequeue(BoundedQueueTest.kt:99)             |            |
```

Comment:
You should've detected spin-lock here.

Reply:
The problem was that previously I didn't decrease `Interleaving.executionPosition`, so after a cycle was detected for the first time we got a lot of new potential switch positions with a big execution number. 
Now `Interleaving` is rolled back every time we find spin-lock so potential switch points stay consistent.